### PR TITLE
feat(dex.trades): add biswap binance

### PR DESCRIPTION
**Pull request by developerfred**
Brief comments on the purpose of your changes:

```
* add biswap on dex.trades
```

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [x] `coin_id` represents the ID of the coin on coinpaprika.com
* [x] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/biswap/bnb/biswap_bnb_trades.sql [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            CASE WHEN amount0In = 0 THEN amount1In ELSE amount0In END    AS token_sold_amount_raw,
+            cast(NULL as double)                                         AS amount_usd,
+            CASE WHEN amount0Out = 0 THEN token1 ELSE token0 END         AS token_bought_address,
+            CASE WHEN amount0In = 0 THEN token1 ELSE token0 END          AS token_sold_address,
+            t.contract_address                                           AS project_contract_address,
+            t.evt_tx_hash                                                AS tx_hash,
+            ''                                                           AS trace_address,
+            t.evt_index
+    FROM {{ source('biswap_bnb', 'BiswapPair_evt_Swap') }} t
+    INNER JOIN {{ source('biswap_bnb', 'BiSwapFactory_evt_PairCreated') }} p
```
Comment (Hosuke) [inline models/biswap/bnb/biswap_bnb_trades.sql:31]: This two tables are actually existed in dune V2 engine, you can add the seeds for test for this PR first.

Comment (developerfred) [reply models/biswap/bnb/biswap_bnb_trades.sql]: done

File: models/biswap/bnb/biswap_bnb_trades.sql [hunk 2]
```diff
@@ -12,7 +12,7 @@
     )
 }}
 
-{% set project_start_date = '2021-03-15' %}
+{% set project_start_date = '2021-05-22' %}     -- select min(evt_block_time) from biswap_bnb.BiswapPair_evt_Swap
```
Comment (Hosuke) [inline models/biswap/bnb/biswap_bnb_trades.sql:15]: Hi @developerfred , I change the project start date based on [the query result](https://dune.com/queries/1647951?d=1). Please verify if any trade record I may miss with this change.

**Review by Hosuke — APPROVED** [review]
To whom it may concern, I have reviewed this PR and here is the summary:

- Metadata & naming conventions look good ✅
- Config blocks match uniqueness tests in schema files ✅
- Tests and seeds added for the new models ✅
- Incremental filters added when joining large tables ✅
- Contract addresses are lowercased ✅
- Inner join to large tables(bnb.transactions) as possible ✅

[Sample query results here](https://dune.com/queries/1647944?d=1)

File: models/biswap/bnb/biswap_bnb_trades.sql (hunk 2 above)
Comment (developerfred) [reply models/biswap/bnb/biswap_bnb_trades.sql:15]: Ok

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**